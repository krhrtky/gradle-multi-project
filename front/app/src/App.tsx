import {Suspense, useState} from 'react'
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import reactLogo from './assets/react.svg'
import viteLogo from '/vite.svg'
import './App.css'
import { useList } from "./libs/api";

const client = new QueryClient();

function App() {
  const [count, setCount] = useState(0)

  return (
    <QueryClientProvider client={client}>
      <div>
        <a href="https://vitejs.dev" target="_blank">
          <img src={viteLogo} className="logo" alt="Vite logo" />
        </a>
        <a href="https://react.dev" target="_blank">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <h1>Vite + React</h1>
      <Component />
      <div className="card">
        <button onClick={() => setCount((count) => count + 1)}>
          count is {count}
        </button>
        <p>
          Edit <code>src/App.tsx</code> and save to test HMR
        </p>
      </div>
      <p className="read-the-docs">
        Click on the Vite and React logos to learn more
      </p>
    </QueryClientProvider>
  )
}

const Component = function () {

    const query = useList();

    return (
        <Suspense fallback={<div>suspended</div>}>
            <div>
                {JSON.stringify(query.data?.data)}
            </div>
        </Suspense>
    );
}

export default App
