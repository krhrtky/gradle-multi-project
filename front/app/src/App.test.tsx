import { describe, it, expect, afterEach } from "vitest";
import { render, cleanup } from "@testing-library/react";
import App from "./App";

describe(App.name, () => {
    afterEach(cleanup);
    it("first render", async () => {
        const { container } = render( <App/>);
        expect(container.innerHTML).toMatchSnapshot();
    });
});
