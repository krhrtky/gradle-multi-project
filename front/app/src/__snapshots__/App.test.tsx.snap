// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > first render 1`] = `"<div><a href=\\"https://vitejs.dev\\" target=\\"_blank\\"><img class=\\"logo\\" alt=\\"Vite logo\\" src=\\"/vite.svg\\"></a><a href=\\"https://react.dev\\" target=\\"_blank\\"><img class=\\"logo react\\" alt=\\"React logo\\" src=\\"/src/assets/react.svg\\"></a></div><h1>Vite + React</h1><div>loading...</div><div class=\\"card\\"><button>count is 0</button><p>Edit <code>src/App.tsx</code> and save to test HMR</p></div><p class=\\"read-the-docs\\">Click on the Vite and React logos to learn more</p>"`;
